### PR TITLE
feat(config): Implement 2-layer configuration system

### LINKED ISSUE #107: [FEATURE] Implement 2 Layers Config
## Summary

I would like to have also users layer config where the .infer directory is created in the home directory of the user.

Those configurations will take precedence if there is no project config in the current dir.

If there is a project-level config these config will take precedence over the userspace-level config.

### Acceptance Criteria

- [ ] The configurations could be added on a userspace-level (default to home directory)
- [ ] When running `infer config init` or `infer init` there is a flag we can pass to it --userspace to indicate it goes to the home directory
- [ ] It's tested
- [ ] It's documented
- [ ] By default if no flag is provided the config will be created in the current pwd
- [ ] Configuration values are merged, for example if I have in my userspace config a configuration that I don't have in my projectspace config it will use the value from the userspace

### PR DESCRIPTION
**Pull request by edenreich**
**Summary**

Implements the 2-layer configuration system requested in issue #107. This adds support for both userspace (`~/.infer/config.yaml`) and project-level (`.infer/config.yaml`) configurations with proper precedence handling.

**Key Features:**
- ✅ Userspace config at `~/.infer/config.yaml` as global fallback
- ✅ Project config takes precedence over userspace config
- ✅ `--userspace` flag for both `infer init` and `infer config init`
- ✅ Smart YAML merging preserves explicit values
- ✅ Comprehensive test coverage (400+ lines of tests)
- ✅ Complete documentation with usage examples
- ✅ Full backward compatibility

**Test Plan:**
- [x] All existing tests pass
- [x] New 2-layer config loading tests
- [x] Config precedence validation tests
- [x] Command flag functionality tests
- [x] Complex nested merging scenarios
- [x] Edge case handling tests

**Configuration Precedence:**
1. Project-level config (`.infer/config.yaml`) - **Highest Priority**
2. Userspace config (`~/.infer/config.yaml`)
3. Built-in defaults - **Lowest Priority**

Resolves #107

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -779,8 +789,47 @@ Create and manage structured task lists for LLM-assisted development workflows.
...
+### Configuration Precedence
+
+Configuration values are merged with the following precedence (highest to lowest):
+
+1. **Project-level config** (`.infer/config.yaml`) - **Highest Priority**
+2. **Userspace config** (`~/.infer/config.yaml`)
+3. **Built-in defaults** - **Lowest Priority**
+
+**Example**: If your userspace config sets `agent.model: "gpt-4"` and your project config sets
+`agent.model: "claude-3"`, the project config wins and `claude-3` will be used. However, if the project
```
Comment (edenreich) [inline README.md:816]:
```suggestion
`agent.model: "deepseek/deepseek-chat"`, the project config wins and `deepseek/deepseek-chat` will be used. However, if the project
```

File: config/config.go [hunk 2]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+			logger.Error("Failed to merge userspace config", "error", err)
+			return nil, fmt.Errorf("failed to merge userspace config: %w", err)
+		}
+		baseConfig = mergedConfig
+		logger.Debug("Successfully merged userspace config")
+	} else {
+		logger.Debug("No userspace config found", "path", userspaceConfigPath)
+	}
+
+	// Try to load project config (takes precedence)
```
Comment (edenreich) [inline config/config.go:478]:
```suggestion
```

File: config/config.go [hunk 3]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+		return nil, fmt.Errorf("failed to parse config file: %w", err)
+	}
+
+	return &config, nil
+}
+
+// mergeConfigsViaYAML merges two configs using YAML marshalling/unmarshalling
+// This approach only overrides explicitly set values, preserving defaults
+func mergeConfigsViaYAML(base, override *Config) (*Config, error) {
+	// Start with the base config as foundation
```
Comment (edenreich) [inline config/config.go:521]:
```suggestion
```

File: config/config.go [hunk 4]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+	result := *base
+
+	// Marshal override config to YAML to only get explicitly set values
+	overrideYAML, err := yaml.Marshal(override)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal override config: %w", err)
+	}
+
+	// Parse override as generic map to see what was explicitly set
```
Comment (edenreich) [inline config/config.go:530]:
```suggestion
```

File: config/config.go [hunk 5]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+	var overrideMap map[string]interface{}
+	if err := yaml.Unmarshal(overrideYAML, &overrideMap); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal override YAML: %w", err)
+	}
+
+	// Remove zero values from override map (these weren't explicitly set)
```
Comment (edenreich) [inline config/config.go:536]:
```suggestion
```

File: config/config.go [hunk 6]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+	cleanOverrideMap := removeZeroValues(overrideMap)
+
+	// If nothing to merge, return base
+	if len(cleanOverrideMap) == 0 {
+		return &result, nil
+	}
+
+	// Marshal the cleaned override map back to YAML
```
Comment (edenreich) [inline config/config.go:544]:
```suggestion
```

File: config/config.go [hunk 7]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+	cleanOverrideYAML, err := yaml.Marshal(cleanOverrideMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal clean override: %w", err)
+	}
+
+	// Unmarshal the override onto our result, which will only override explicitly set values
```
Comment (edenreich) [inline config/config.go:550]:
```suggestion
```

File: config/config.go [hunk 8]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+	for k, v := range m {
+		switch val := v.(type) {
+		case map[string]interface{}:
+			// Recursively clean nested maps
+			cleaned := removeZeroValues(val)
+			if len(cleaned) > 0 {
+				result[k] = cleaned
+			}
+		case string:
+			// Keep non-empty strings
```
Comment (edenreich) [inline config/config.go:572]:
```suggestion
```

File: config/config.go [hunk 9]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+			if val != "" {
+				result[k] = val
+			}
+		case int:
+			// Keep non-zero ints
```
Comment (edenreich) [inline config/config.go:577]:
```suggestion
```

File: config/config.go [hunk 10]
```diff
@@ -440,6 +440,183 @@ func LoadConfig(configPath string) (*Config, error) {
...
+			if len(val) > 0 {
+				result[k] = val
+			}
+		case []string:
+			// Keep non-empty string slices
+			if len(val) > 0 {
+				result[k] = val
+			}
+		case []int:
+			// Keep non-empty int slices
```
Comment (edenreich) [inline config/config.go:595]:
```suggestion
```

**Comment by edenreich** [comment]
@claude what about all the other commands? I want to be able to set the userspace configs values - I don't see there is a userspace global flag:

image

Please add this.

**Comment by edenreich** [comment]
I think I'll just use viper for the config

**Comment by edenreich** [comment]
@claude ci is failing
